Applications/Spire/Source/Ui/IconButton.cpp

```diff
@@ -64,27 +62,29 @@ void IconButton::keyPressEvent(QKeyEvent* event) {
 
 void IconButton::paintEvent(QPaintEvent* event) {
   auto painter = QPainter(this);
-  if(!underMouse() || !isEnabled()) {
-    painter.fillRect(rect(), m_style.m_default_background_color);
-  } else {
-    painter.fillRect(rect(), m_style.m_hover_background_color);
-  }
+  painter.fillRect(rect(), get_background_color());
   auto icon = QPixmap::fromImage(m_icon);
   auto image_painter = QPainter(&icon);
   image_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
   image_painter.fillRect(icon.rect(), get_current_icon_color());
   painter.drawPixmap((width() - icon.width()) / 2,
     (height() - icon.height()) / 2, icon);
   if(hasFocus()) {
-    painter.setPen({QColor("#4B23A0"), static_cast<double>(scale_width(1))});
-    painter.drawRect(rect().adjusted(0, 0, -scale_width(1), -scale_height(1)));
+    draw_border(rect(), "#4B23A0", &painter);
```

Comment:
Why is this color not put in IconButton::Style:?

Reply:
For now it's essentially a constant. I don't see a reason why we couldn't add it to Style, but until there's a need for I'll leave it out.

Reply:
I don’t understand. The background color can be specified through passing IconButton::Style to the constructor when creating an Iconbutton, but why is the border color a constant?

Reply:
The border color is currently a constant because there's no instances that need a different border color. It's likely that the font selector's buttons will need options to customize this, but we generally only add things as they're required.

In this case adding a member to the Style struct with a default value is trivial, so I don't think leaving it out until it's needed is a big deal.